src/python/build_management/build_manager.py

```diff
@@ -219,7 +223,7 @@ def _unpack_build(base_build_dir, build_dir, build_url, target_weights=None):
     return False
 
   unpack_everything = environment.get_value('UNPACK_ALL_FUZZ_TARGETS_AND_FILES')
-  if not unpack_everything:
+  if not unpack_everything and not is_auxiliary_build:
```

Comment:
If we always unpack everything, there will be disk issues. Let's change the logic of _set_random_fuzz_target_for_fuzzing_if_needed to use the existing FUZZ_TARGET, if it exists. No need for this is_auxiliary_build argument.

Reply:
My intention was to never unpack everything for auxiliary builds, as the fuzz target must be chosen before that. I even had an explicit assertion for that, but ended up removing it when was writing the tests. So, right now the code would not exit and would unpack everything instead, but I don't think that might happen in reality.

Reply:
Ok I see, _get_file_match_callback() will still return the right matcher.

In this case I think we can remove this auxiliary argument, and simply change the logic of _set_random_fuzz_target_for_fuzzing_if_needed to do nothing if FUZZ_TARGET is already set.